### PR TITLE
`StoreTransaction`: added deprecations for version 3.x properties

### PR DESCRIPTION
**Pull request by NachoSoto**
Fixes [CF-232] and [CF-231].

[CF-232]: https://revenuecats.atlassian.net/browse/CF-232?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CF-231]: https://revenuecats.atlassian.net/browse/CF-231?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by aboedo — APPROVED** [review]
❤️

**Comment by NachoSoto** [comment]
@aboedo I added the change to the migration guide here too.

File: Documentation.docc/V4_API_Migration_guide.md [hunk 1]
```diff
@@ -134,6 +134,8 @@ These types replace native StoreKit types in all public API methods that used th
 | ------------ | ------------------------------------- | 
 | RCPurchaserInfo | RCCustomerInfo |
 | RCTransaction | RCStoreTransaction |
+| RCTransaction.productId | RCStoreTransaction.productIdentifier |
```
Comment (aboedo) [inline Documentation.docc/V4_API_Migration_guide.md:137]: seeing this now I'm surprised we didn't do backticks for any of them

**Review by aboedo — APPROVED** [review]

File: Documentation.docc/V4_API_Migration_guide.md (hunk 1 above)
Comment (NachoSoto) [reply Documentation.docc/V4_API_Migration_guide.md:137]: Single ones you mean?

**Pull request merged**